lib/cache/memory-cache-store.js

```diff
@@ -0,0 +1,417 @@
+'use strict'
+
+const { Writable, Readable } = require('node:stream')
+
+/**
+ * @typedef {import('../../types/cache-interceptor.d.ts').default.CacheStore} CacheStore
+ * @implements {CacheStore}
+ *
+ * @typedef {{
+ *  readers: number
+ *  readLock: boolean
+ *  writeLock: boolean
+ *  opts: import('../../types/cache-interceptor.d.ts').default.CacheStoreValue
+ *  body: Buffer[]
+ * }} MemoryStoreValue
+ */
+class MemoryCacheStore {
+  #maxEntries = Infinity
+
+  #maxEntrySize = Infinity
+
+  /**
+   * @type {((err) => void) | undefined}
+   */
+  #errorCallback = undefined
+
+  #entryCount = 0
+
+  /**
+   * @type {Map<string, Map<string, MemoryStoreValue>>}
+   */
+  #data = new Map()
+
+  /**
+   * @param {import('../../types/cache-interceptor.d.ts').default.MemoryCacheStoreOpts | undefined} opts
+   */
```

Comment:
opts is not mandatory, right?


```suggestion
   * @param {import('../../types/cache-interceptor.d.ts').default.MemoryCacheStoreOpts | undefined} [opts]
```